packages/babel-plugin-minify-dead-code-elimination/src/index.js

```diff
@@ -710,6 +715,10 @@ module.exports = ({ types: t, traverse }) => {
           traverse.clearCache();
           path.scope.crawl();
 
+          if (!isEvalScopesMarked(path.scope)) {
+            markEvalScopes(path);
```

Comment:
I'd love to merge this for now.

Just one suggestion — let's move `isEvalScopesMarked` into `markEvalScopes` (checking it within and returning if scope is already marked). That way API becomes simpler (`markEvalScopes(path)`), there's less vars to exports, etc.

Another thing to consider is that `markEvalScopes` operates on path, and `hasEval` operates on scope. In an ideal world, that means these methods belong to those objects — `path.markScopesWithEval`, `scope.hasDirectEvals`.